crates/router/src/core/errors.rs

```diff
@@ -50,6 +50,31 @@ macro_rules! impl_error_display {
     };
 }
 
+#[macro_export]
+macro_rules! not_supported {
```

Comment:
@swangi-kumari Let's create diff macro_rules for different purposes, rename this to capture_method_not_supported. and handle error related to capture_methods